src/array.cr

```diff
@@ -1630,15 +1648,31 @@ class Array(T)
     shift { nil }
   end
 
+  # Returns an array with all the elements in the collection randomized
+  # using the given *random* number generator.
   def shuffle(random = Random::DEFAULT)
     dup.shuffle!(random)
   end
 
+  # Modifies `self` by randomizing the order of elements in the collection
+  # using the given *random* number generator.
   def shuffle!(random = Random::DEFAULT)
     @buffer.shuffle!(size, random)
     self
   end
 
+  # Returns an array with all the elements in the collection sorted using *quicksort*.
```

Comment:
I don't know if I would mention the sorting algorithm. I wouldn't mention `Array.quicksort!`, it's just an implementation detail and it's not public API (we should mark it as protected).